### ISSUE TITLE
VP 2.2 with EUI has missing UI features

### ISSUE THREAD

**Issue by punger** [LoneGazebo/Community-Patch-DLL#8873]
_1. Mod version, and DirectX version if reporting a graphical issue (X.Y.Z, e.g. 2.2.0, DX 10 & 11):_
2.2 

_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description:_ 
I am setting up VP 2.2 on a new install of civ 5 on a new machine. When I install VP in any flavor that includes the EUI a number of strange UI symptoms occur: This happens with no mods enabled at all.
1. In options I cannot click on the Accept button. It does nothing. Clicking on Cancel makes the dialog go away and it seems to save the settings anyway. For a while at least.
2. More importantly, though, the box on the lower left to interact with the currently selected unit is absent.
3. Units, even if selected, do not show the slowly rotating yellow circle that indicates that they are selected.
4. I do not appear to be able to select units using the mouse.
There are some other missing pieces, but they may be merely consequences of the above.

Using the keyboard shortcuts, I can cycle through units and use the shortcuts or mouse to make them do things.

When I install VP in any of the configurations that does not include EUI, the above problems do not occur

_4. Steps to reproduce (optional):_
Just start a game

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. For instructions, go to the repository's main page, under "To enable logging for bug reports"):

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves; you can change autosave frequency in the game's Options menu):

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Comment by punger** [comment]
I installed EUI without any of the VP mods and it works fine.  It is an incompatible (the most recent) version so it doesn't address the problem.  The problem seems to appear with any VP 2.2 installation that includes EUI.  Both pieces together are sufficient and necessary.

**Comment by RecursiveVision** [comment]
Need Lua.log. See here for instructions on how to enable logging: https://github.com/LoneGazebo/Community-Patch-DLL#to-enable-logging-for-bug-reports

**Comment by punger** [comment]
[Lua.log](https://github.com/LoneGazebo/Community-Patch-DLL/files/8694224/Lua.log)
After looking at this, I went looking for OptionsMenu.lua and I see two of them on my system.

I would upload them, but the type is not accepted.  In any event, they are different.  One in ...\Assets\UI\Options starts like:

```
-------------------------------------------------
-- Options Menu
-------------------------------------------------
local m_FullscreenResList = {};
local m_iResolutionCount;
local m_maxX, m_maxY = OptionsManager.GetMaxResolution();

local m_bInGameQuickCombatState_Cached = false;
local m_bInGameQuickMovementState_Cached = false;
...

```
The one in ...\Assets\UI\Options\UI_bc1 starts like
```
-------------------------------------------------
-- Options Menu
-- modified by bc1 from Civ V 1.0.3.276 code
-- add new options for EUI
-------------------------------------------------
--print("Loading EUI option menu...",os.clock())
local ipairs = ipairs
local math_floor = math.floor
local string_format = string.format
local table_insert = table.insert

local EUI_options = Modding.OpenUserData( "Enhanced User Interface Options", 1);
...
```

The code around line 634 (lines 630 - 648) are
```
----------------------------------------------------------------
-- Display updating
----------------------------------------------------------------

function UpdateGameOptionsDisplay()

	local bIsInGame = (ContextPtr:GetID() == "OptionsMenu_InGame");
	
    --Controls.NoCitizenWarningCheckbox:SetCheck( OptionsManager.IsNoCitizenWarning() );
    Controls.AutoWorkersDontReplaceCB:SetCheck( OptionsManager.IsAutoWorkersDontReplace_Cached() );
    Controls.AutoWorkersDontRemoveFeaturesCB:SetCheck( OptionsManager.IsAutoWorkersDontRemoveFeatures_Cached() );
    Controls.NoRewardPopupsCheckbox:SetCheck( OptionsManager.IsNoRewardPopups_Cached() );
    Controls.NoBasicHelpCheckbox:SetCheck( OptionsManager.IsNoBasicHelp_Cached() );
    Controls.NoTileRecommendationsCheckbox:SetCheck( OptionsManager.IsNoTileRecommendations_Cached() );
    Controls.CivilianYieldsCheckbox:SetCheck( OptionsManager.IsCivilianYields_Cached() );
    Controls.SinglePlayerAutoEndTurnCheckBox:SetCheck(OptionsManager.GetSinglePlayerAutoEndTurnEnabled_Cached());
		Controls.MultiplayerAutoEndTurnCheckbox:SetCheck(OptionsManager.GetMultiplayerAutoEndTurnEnabled_Cached());
    Controls.QuickSelectionAdvCheckbox:SetCheck( OptionsManager.GetQuickSelectionAdvanceEnabled_Cached() );

```

**Comment by nobubblegums** [comment]
Most probably the VP installer misplaced EUI (ui_bc1 folder). You can download VP's EUI from here https://drive.google.com/drive/folders/17YEFQN1HAxfHsmlFks_FfBzoLRw5vLb3?usp=sharing

Place it to "steamapps\common\Sid Meier's Civilization V\Assets\DLC" , be sure to remove the EUI version you installed.

**Comment by punger** [comment]
That did not work, assuming I followed your directions correctly.

The VP installer places the UI_bc1 folder in the place you mention.  If I delete that folder and replace it with what you had me download, not only is EUI not operational -- legal notices pop up, the Options dialog shows only original settings, and so on -- but when I try to run a game, there are a bunch of popups that indicate missing resources.  The game is unplayable.

**Comment by punger** [comment]
Ah, also my initial report had the paths wrong.  This is the correct path,

C:\Program Files (x86)\Steam\steamapps\common\Sid Meier's Civilization V\Assets\DLC\UI_bc1\Options\OptionsMenu.lua

Bit of path segment dyslexia there.  If that was what suggested the bad path for install, I apologize for being sloppy.

**Comment by ramy-gendy** [comment]
I am having the same issue, with v2.2 of VP. When I start the a game, I get a bunch of popups telling me about missing textures.

**Comment by ramy-gendy** [comment]
> Ah, also my initial report had the paths wrong. This is the correct path,
> 
> C:\Program Files (x86)\Steam\steamapps\common\Sid Meier's Civilization V\Assets\DLC\UI_bc1\Options\OptionsMenu.lua
> 
> Bit of path segment dyslexia there. If that was what suggested the bad path for install, I apologize for being sloppy.

By any chance do you have the game installed on a different hard drive than C:/?

**Comment by punger** [comment]
No, it is on C.  One potential wrinkle is that my Documents folder is actually mapped to a OneDrive folder on this machine.

**Comment by nobubblegums** [comment]
Sorry, I can't really think of anything else if everything is properly placed. Could be an issue with permissions/antivirus preventing some filled from getting installed and such.

Next version will ask the path for Civ5 when installing EUI, hopefully that may help someone.
